android/app/src/main/java/app/organicmaps/util/SharingUtils.java

```diff
@@ -92,7 +103,28 @@ public static void shareBookmark(@NonNull Context context, @NonNull BookmarkInfo
     context.startActivity(Intent.createChooser(intent, context.getString(R.string.share)));
   }
 
-  public static void shareBookmarkFile(Context context, String fileName)
+  public static ActivityResultLauncher<Intent> RegisterLauncher(@NonNull Fragment fragment)
+  {
+    return fragment.registerForActivityResult(
+            new ActivityResultContracts.StartActivityForResult(), result -> {
+              if (result.getResultCode() == Activity.RESULT_OK && result.getData() != null)
+              {
+                Uri destinationUri = result.getData().getData();
+                Uri sourceUri = sourceFileUri;
+                sourceFileUri = null;
+                try
+                {
+                  if (sourceUri != null && destinationUri != null)
+                    StorageUtils.copyFile(fragment.requireContext().getContentResolver(), sourceUri, destinationUri);
+
+                } catch (IOException e)
```

Comment:
```suggestion
                }
                catch (IOException e)
```

Reply:
If in doubt, check other code for examples.

Reply:
fixed
thank you very much for review)